wails 2.9.0 linux issue

### Description

I can't start dev and I can't build

### To Reproduce

try starting dev or building on linux latest version of wails

### Expected behaviour

it shouldn't give error

### Screenshots

![image](https://github.com/wailsapp/wails/assets/87276663/3a4d3ec5-dee6-4456-890f-0ba7e2dbbcae)


### Attempted Fixes

_No response_

### System Details

```shell
Wails Doctor



# Wails
Version         | v2.9.0
Package Manager | pacman

# System
┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
| OS           | EndeavourOS                                                                                          |
| Version      | Unknown                                                                                              |
| ID           | endeavouros                                                                                          |
| Go Version   | go1.22.4                                                                                             |
| Platform     | linux                                                                                                |
| Architecture | amd64                                                                                                |
| CPU          | Intel(R) Core(TM) i3-4005U CPU @ 1.70GHz                                                             |
| GPU 1        | GF117M [GeForce 610M/710M/810M/820M / GT 620M/625M/630M/720M] (NVIDIA Corporation) - Driver: nouveau |
| GPU 2        | Haswell-ULT Integrated Graphics Controller (Intel Corporation) - Driver: i915                        |
| Memory       | 4GB                                                                                                  |
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

# Dependencies
┌───────────────────────────────────────────────────────────────────┐
| Dependency | Package Name | Status    | Version                   |
| *docker    | docker       | Available | 1:26.1.4-1                |
| gcc        | gcc          | Installed | 14.1.1+r58+gfc9fb69ad62-1 |
| libgtk-3   | gtk3         | Installed | 1:3.24.42-1               |
| libwebkit  | webkit2gtk   | Installed | 2.44.2-1                  |
| npm        | npm          | Available | 10.8.1-1                  |
| pkg-config | pkgconf      | Installed | 2.1.1-1                   |
└───────────────────── * - Optional Dependency ─────────────────────┘

# Diagnosis
Required package(s) installation details:
  - npm: sudo pacman -S npm

Optional package(s) installation details:
  - docker: sudo pacman -S docker

 WARNING  Your system has missing dependencies!

 ♥   If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony
```


### Additional context

I'm using pnpm to install packages and that's why I don't have npm

For completeness, here is my error log and doctor output:

```
  • Compiling application: # github.com/wailsapp/wails/v2/internal/frontend/desktop/linux
In file included from /usr/include/glib-2.0/gobject/gobject.h:26,
                 from /usr/include/glib-2.0/gobject/gbinding.h:31,
                 from /usr/include/glib-2.0/glib-object.h:24,
                 from /usr/include/glib-2.0/gio/gioenums.h:30,
                 from /usr/include/glib-2.0/gio/giotypes.h:30,
                 from /usr/include/glib-2.0/gio/gio.h:28,
                 from /usr/include/gtk-3.0/gdk/gdkapplaunchcontext.h:28,
                 from /usr/include/gtk-3.0/gdk/gdk.h:32,
                 from /usr/include/gtk-3.0/gtk/gtk.h:30,
                 from window.c:2:
window.c: In function ‘SetupWebview’:
/usr/include/glib-2.0/gobject/gtype.h:2656:42: error: passing argument 1 of ‘gtk_drag_dest_unset’ from incompatible pointer type [-Wincompatible-pointer-types]
 2656 | #  define _G_TYPE_CIC(ip, gt, ct)       ((ct*) (void *) ip)
/usr/include/glib-2.0/gobject/gtype.h:528:66: note: in expansion of macro ‘_G_TYPE_CIC’
  528 | #define G_TYPE_CHECK_INSTANCE_CAST(instance, g_type, c_type)    (_G_TYPE_CIC ((instance), (g_type), c_type))
      |                                                                  ^~~~~~~~~~~
/usr/include/glib-2.0/gobject/gobject.h:54:38: note: in expansion of macro ‘G_TYPE_CHECK_INSTANCE_CAST’
   54 | #define G_OBJECT(object)            (G_TYPE_CHECK_INSTANCE_CAST ((object), G_TYPE_OBJECT, GObject))
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~
window.c:510:29: note: in expansion of macro ‘G_OBJECT’
  510 |         gtk_drag_dest_unset(G_OBJECT(webview));
      |                             ^~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:87:
/usr/include/gtk-3.0/gtk/gtkdragdest.h:83:47: note: expected ‘GtkWidget *’ {aka ‘struct _GtkWidget *’} but argument is of type ‘GObject *’ {aka ‘struct _GObject *’}
   83 | void gtk_drag_dest_unset (GtkWidget          *widget);
      |                           ~~~~~~~~~~~~~~~~~~~~^~~~~~
```

```
# Wails
Version         | v2.9.0
Package Manager | pacman

# System
┌───────────────────────────────────────────────────────────────────────────────────────┐
| OS           | Arch Linux                                                             |
| Version      | Unknown                                                                |
| ID           | arch                                                                   |
| Go Version   | go1.22.3                                                               |
| Platform     | linux                                                                  |
| Architecture | amd64                                                                  |
| CPU          | 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz                         |
| GPU          | TigerLake-LP GT2 [Iris Xe Graphics] (Intel Corporation) - Driver: i915 |
| Memory       | 15GB                                                                   |
└───────────────────────────────────────────────────────────────────────────────────────┘

# Dependencies
┌───────────────────────────────────────────────────────────────────┐
| Dependency | Package Name | Status    | Version                   |
| *docker    | docker       | Installed | 1:26.1.3-1                |
| gcc        | gcc          | Installed | 14.1.1+r58+gfc9fb69ad62-1 |
| libgtk-3   | gtk3         | Installed | 1:3.24.42-1               |
| libwebkit  | webkit2gtk   | Installed | 2.44.2-1                  |
| npm        | npm          | Installed | 10.8.0-1                  |
| pkg-config | pkgconf      | Installed | 2.1.1-1                   |
└───────────────────── * - Optional Dependency ─────────────────────┘

# Diagnosis
 SUCCESS  Your system is ready for Wails development!
```

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -48,7 +48,7 @@ jobs:
       - name: Run tests (Ubuntu 24.04)
         if: matrix.os == 'ubuntu-24.04'
         working-directory: ./v2
-        run: go test -v ./...-tags webkit2_41
+        run: go test -v ./... -tags webkit2_41
 
   test_js:
     name: Run JS Tests

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -12,7 +12,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest, macos-11]
+        os: [ubuntu-22.04, windows-latest, macos-latest, macos-11]
         go-version: ['1.21']
 
     steps:
@@ -77,7 +77,7 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest, ubuntu-24.04]
+        os: [ubuntu-22.04, windows-latest, macos-latest, ubuntu-24.04]
         template:
           [
             svelte,

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -48,7 +48,7 @@ jobs:
       - name: Run tests (Ubuntu 24.04)
         if: matrix.os == 'ubuntu-24.04'
         working-directory: ./v2
-        run: go test -v ./... -tags webkit2_41
+        run: go test -v -tags webkit2_41 ./...
 
   test_js:
     name: Run JS Tests

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -19,13 +19,17 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Install linux dependencies ( 22.04 )
+      - uses: awalsh128/cache-apt-pkgs-action@latest
         if: matrix.os == 'ubuntu-22.04'
-        run: sudo apt-get update -y && sudo apt-get install -y libgtk-3-dev libwebkit2gtk-4.0-dev build-essential pkg-config
+        with:
+          packages: libgtk-3-dev libwebkit2gtk-4.0-dev build-essential pkg-config
+          version: 1.0
 
-      - name: Install linux dependencies ( 24.04 )
+      - uses: awalsh128/cache-apt-pkgs-action@latest
         if: matrix.os == 'ubuntu-24.04'
-        run: sudo apt-get update -y && sudo apt-get install -y libgtk-3-dev libwebkit2gtk-4.1-dev build-essential pkg-config
+        with:
+          packages: libgtk-3-dev libwebkit2gtk-4.1-dev build-essential pkg-config
+          version: 1.0
 
       - name: Setup Go
         uses: actions/setup-go@v4

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -124,5 +124,15 @@ jobs:
           mkdir -p ./test-${{ matrix.template }}
           cd ./test-${{ matrix.template }}
           wails init -n ${{ matrix.template }} -t ${{ matrix.template }} -ci
+
+      - name: Build template '${{ matrix.template }}'
+        if: matrix.os != 'ubuntu-24.04'
+        run: |
           cd ${{ matrix.template }}
           wails build -v 2
+
+      - name: Build template '${{ matrix.template }}' (Ubuntu 24.04)
+        if: matrix.os == 'ubuntu-24.04'
+        run: |
+          cd ${{ matrix.template }}
+          wails build -v 2 -tags webkit2_41

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -12,7 +12,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-22.04, windows-latest, macos-latest, macos-11]
+        os: [ubuntu-22.04, ubuntu-24.04, windows-latest, macos-latest, macos-11]
         go-version: ['1.21']
 
     steps:
@@ -77,7 +77,7 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        os: [ubuntu-22.04, windows-latest, macos-latest, ubuntu-24.04]
+        os: [ubuntu-22.04, windows-latest, macos-latest, ubuntu-24.04, macos-11]
         template:
           [
             svelte,

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -124,20 +124,21 @@ jobs:
         if: matrix.os == 'ubuntu-24.04'
         run: sudo apt-get update -y && sudo apt-get install -y libgtk-3-dev libwebkit2gtk-4.1-dev build-essential pkg-config
 
-      - name: Generate template '${{ matrix.template }}'
+      - name: Generate & Build template '${{ matrix.template }}'
+        if: matrix.os != 'ubuntu-24.04'
         run: |
           mkdir -p ./test-${{ matrix.template }}
           cd ./test-${{ matrix.template }}
           wails init -n ${{ matrix.template }} -t ${{ matrix.template }} -ci
-
-      - name: Build template '${{ matrix.template }}'
-        if: matrix.os != 'ubuntu-24.04'
-        run: |
           cd ${{ matrix.template }}
           wails build -v 2
 
-      - name: Build template '${{ matrix.template }}' (Ubuntu 24.04)
+      - name: Generate & Build template '${{ matrix.template }}' (ubuntu-24.04)
         if: matrix.os == 'ubuntu-24.04'
         run: |
+          mkdir -p ./test-${{ matrix.template }}
+          cd ./test-${{ matrix.template }}
+          wails init -n ${{ matrix.template }} -t ${{ matrix.template }} -ci
           cd ${{ matrix.template }}
           wails build -v 2 -tags webkit2_41
+

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -41,9 +41,14 @@ jobs:
         run: go test -v ./...
 
       - name: Run tests (!mac)
-        if: matrix.os != 'macos-latest' && matrix.os != 'macos-11'
+        if: matrix.os != 'macos-latest' && matrix.os != 'macos-11' && matrix.os != 'ubuntu-24.04'
         working-directory: ./v2
         run: go test -v ./...
+
+      - name: Run tests (Ubuntu 24.04)
+        if: matrix.os == 'ubuntu-24.04'
+        working-directory: ./v2
+        run: go test -v ./...-tags webkit2_41
 
   test_js:
     name: Run JS Tests

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -19,9 +19,13 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Install linux dependencies
-        if: matrix.os == 'ubuntu-latest'
+      - name: Install linux dependencies ( 22.04 )
+        if: matrix.os == 'ubuntu-22.04'
         run: sudo apt-get update -y && sudo apt-get install -y libgtk-3-dev libwebkit2gtk-4.0-dev build-essential pkg-config
+
+      - name: Install linux dependencies ( 24.04 )
+        if: matrix.os == 'ubuntu-24.04'
+        run: sudo apt-get update -y && sudo apt-get install -y libgtk-3-dev libwebkit2gtk-4.1-dev build-essential pkg-config
 
       - name: Setup Go
         uses: actions/setup-go@v4
@@ -73,7 +77,7 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest]
+        os: [ubuntu-latest, windows-latest, macos-latest, ubuntu-24.04]
         template:
           [
             svelte,
@@ -107,9 +111,13 @@ jobs:
           go install
           wails -help
 
-      - name: Install linux dependencies
-        if: matrix.os == 'ubuntu-latest'
+      - name: Install linux dependencies ( 22.04 )
+        if: matrix.os == 'ubuntu-22.04'
         run: sudo apt-get update -y && sudo apt-get install -y libgtk-3-dev libwebkit2gtk-4.0-dev build-essential pkg-config
+
+      - name: Install linux dependencies ( 24.04 )
+        if: matrix.os == 'ubuntu-24.04'
+        run: sudo apt-get update -y && sudo apt-get install -y libgtk-3-dev libwebkit2gtk-4.1-dev build-essential pkg-config
 
       - name: Generate template '${{ matrix.template }}'
         run: |

--- a/.github/workflows/pr.yml
+++ b/.github/workflows/pr.yml
@@ -56,7 +56,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-22.04, windows-latest, macos-latest, ubuntu-24.04]
-        go-version: ['1.22']
+        go-version: ['1.21']
 
     steps:
       - name: Checkout code

--- a/.github/workflows/pr.yml
+++ b/.github/workflows/pr.yml
@@ -62,9 +62,13 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v3
 
-      - name: Install linux dependencies
-        if: matrix.os == 'ubuntu-latest'
+      - name: Install linux dependencies ( 22.04 )
+        if: matrix.os == 'ubuntu-22.04'
         run: sudo apt-get update -y && sudo apt-get install -y libgtk-3-dev libwebkit2gtk-4.0-dev build-essential pkg-config
+
+      - name: Install linux dependencies ( 24.04 )
+        if: matrix.os == 'ubuntu-24.04'
+        run: sudo apt-get update -y && sudo apt-get install -y libgtk-3-dev libwebkit2gtk-4.1-dev build-essential pkg-config
 
       - name: Setup Go
         uses: actions/setup-go@v3

--- a/.github/workflows/pr.yml
+++ b/.github/workflows/pr.yml
@@ -83,6 +83,11 @@ jobs:
         run: go test -v ./...
 
       - name: Run tests (!mac)
-        if: matrix.os != 'macos-latest'
+        if: matrix.os != 'macos-latest' && matrix.os != 'ubuntu-24.04'
         working-directory: ./v2
         run: go test -v ./...
+
+      - name: Run tests (Ubuntu 24.04)
+        if: matrix.os == 'ubuntu-24.04'
+        working-directory: ./v2
+        run: go test -v -tags webkit2_41 ./...

--- a/.github/workflows/pr.yml
+++ b/.github/workflows/pr.yml
@@ -55,7 +55,7 @@ jobs:
     if: github.event.review.state == 'approved'
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest]
+        os: [ubuntu-22.04, windows-latest, macos-latest, ubuntu-24.04]
         go-version: ['1.22']
 
     steps:

--- a/v2/internal/frontend/desktop/linux/window.c
+++ b/v2/internal/frontend/desktop/linux/window.c
@@ -507,7 +507,7 @@ GtkWidget *SetupWebview(void *contentManager, GtkWindow *window, int hideWindowO
 
     if(disableWebViewDragAndDrop)
     {
-        gtk_drag_dest_unset(G_OBJECT(webview));
+        gtk_drag_dest_unset(webview);
     }
 
     if(enableDragAndDrop)

